### PR TITLE
`Programming exercises`: Fix broken results in problem statement in the instructor detail page

### LINKED ISSUE #7880: Problem statement in the programming exercise detail page is not showing the template repo status correctly
### Describe the bug

The problem statement on the detail page of programming exercises no longer properly displays with the status of the latest template submission. Instead it sometimes shows question marks for everything:
image
Or it shows it correctly in the task list at the top, but not in the rest of the problem statement:
image

Considering the inconsistent behavior I'd assume that this was not an intentional change and was likely introduced in #7632, as the bug is visible in the screencasts there.

### To Reproduce

1. Go to any programming exercise detail page
2. See the described issue

### Expected behavior

The problem statement on the detail page previously showed it like this:
image
This is the status of the latest template repo submission in that exercise. As you can see the task bar and the tasks have the proper markings that they failed

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

develop

### What browsers are you seeing the problem on?

Chrome

### Additional context

 I specifically was on TS9 and logged in with an admin account using a Chromium based browser

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by milljoniaer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I added multiple screenshots/screencasts of my UI changes.
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 2 (Jenkins and Gitlab).

### Motivation and Context
The programming exercise problem statement did not include the latest results on the detail pages, since the rendering through angular changed with the detail page component. This was fixed by reinitializing the testCases after changes.
closes #7880 

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to a programming detail page
3. Verify that the problem statement includes the latest results of the template repository
4. Submit something from the student view
5. Verify that the results are shown to the student

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots
image

## Summary by CodeRabbit

- **Refactor**
	- Improved the loading of test cases in the programming exercise instructions to ensure they are only fetched for users with at least tutor access.

### CODE REVIEW

**Review by MaximilianJG — APPROVED** [review]
Tested on Ts5. Works as expected. Results are shown to student! 

Screenshot 2024-01-15 at 16 40 17

**Review by xHadie — APPROVED** [review]
_Manually tested on TS11_Legacy_
Seems to be working fine now:

image

After solving a task:
image

**Review by egekurt123 — DISMISSED** [review]
Manually tested on ts5, works as expected

File: src/main/webapp/app/exercises/programming/shared/instructions-render/programming-exercise-instruction.component.ts [hunk 1]
```diff
@@ -96,6 +96,16 @@ export class ProgrammingExerciseInstructionComponent implements OnChanges, OnIni
      * @param changes
      */
     public ngOnChanges(changes: SimpleChanges) {
+        if (this.exercise?.isAtLeastTutor) {
+            this.testCasesSubscription = this.programmingExerciseGradingService
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/programming/shared/instructions-render/programming-exercise-instruction.component.ts:100]: This opens a new subscription on changes, but does not destroy the old one if it's still present. Can you add a check and unsubscribe if necessary.

**Review by konrad2002 — APPROVED** [review]
manually tested on legacy ts1, works as expected

**Review by LenaKahle — APPROVED** [review]
Approve, tested on ts11.

**Review by Strohgelaender — APPROVED** [review]
code lgtm, tested on TS5 and everything works.

**Review by egekurt123 — APPROVED** [review]
Reapprove after minor changes

**Review by reschandreas — APPROVED** [review]
tested on TS5 works as expected, lgtm

**Review by LenaKahle — APPROVED** [review]
Reapprove.

**Review by mateusmm01 — APPROVED** [review]
Code lgtm 👍

**Pull request merged**